app/javascript/onboarding/components/EmailPreferencesForm.jsx

```diff
@@ -53,53 +53,55 @@ class EmailPreferencesForm extends Component {
     const { email_newsletter, email_digest_periodic } = this.state;
     const { prev, slidesCount, currentSlideIndex } = this.props;
     return (
-      <div className="onboarding-main">
-        <Navigation
-          prev={prev}
-          next={this.onSubmit}
-          slidesCount={slidesCount}
-          currentSlideIndex={currentSlideIndex}
-        />
-        <div className="onboarding-content terms-and-conditions-wrapper">
-          <header className="onboarding-content-header">
-            <h1 className="title">Almost there!</h1>
-            <h2 className="subtitle">
-              Review your email preferences before we continue.
-            </h2>
-          </header>
+      <div className="onboarding-main crayons-modal">
```

Comment:
Any reason the `<Modal />` component is not being used for onboarding? I'm guessing you just want the styling of a modal for onboarding? The same question applies to all the onboarding slides.

Reply:
Looking at how you're using it, I now understand why you did it this way. However, there is a way to reuse the same `<Modal />` component, even in onboarding. Instead of forcing a title prop to be a string, it could be a string or component, allowing the flexibility to render whatever in the title area of the `<Modal /> component. We can go over this tomorrow if you'd like.

For more information on this, check out the [Slot Pattern](https://daveceddia.com/pluggable-slots-in-react-components/) in (P)React.

Reply:
Onboarding was created while ago and I didn't want to update it right away with this PR. This PR is only meant to introduce modal as a component, but not really implementing it anywhere in particular – I have only updated onboarding frontend just a tiny bit so it doesn't look broken.

I think modals also have some logic and JS wired that I don't want to "unwire" right now :D

But eventually - 100% agree, onboarding should be based on the Modal component.